zbus 5.13.1 fails to compile on Windows when tokio feature is enabled

The build fails on Windows because of a type mismatch in src/connection/socket/tcp.rs. It appears the tokio implementation of Task::spawn_blocking returns a nested Result, but the code expects a flat Result<ConnectionCredentials, Error>.

```
   Compiling zbus v5.13.1
error[E0308]: mismatched types
   --> C:\Users\runneradmin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\zbus-5.13.1\src\connection\socket\tcp.rs:120:9
    |
120 | /         crate::Task::spawn_blocking(
121 | |             move || win32_credentials_from_addr(&peer_addr),
122 | |             "peer credentials",
123 | |         )
124 | |         .await
    | |______________^ expected `Result<ConnectionCredentials, Error>`, found `Result<Result<..., ...>, ...>`
    |
    = note: expected enum `std::result::Result<ConnectionCredentials, _>`
               found enum `std::result::Result<std::result::Result<ConnectionCredentials, std::io::Error>, _>`
help: consider using `Result::expect` to unwrap the `std::result::Result<std::result::Result<ConnectionCredentials, std::io::Error>, std::io::Error>` value, panicking if the value is a `Result::Err`
    |
124 |         .await.expect("REASON")
    |               +++++++++++++++++

error[E0308]: mismatched types
   --> C:\Users\runneradmin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\zbus-5.13.1\src\connection\socket\tcp.rs:161:9
    |
161 | /         crate::Task::spawn_blocking(
162 | |             move || win32_credentials_from_addr(&peer_addr),
163 | |             "peer credentials",
164 | |         )
165 | |         .await
    | |______________^ expected `Result<ConnectionCredentials, Error>`, found `Result<Result<..., ...>, ...>`
    |
    = note: expected enum `std::result::Result<ConnectionCredentials, _>`
               found enum `std::result::Result<std::result::Result<ConnectionCredentials, std::io::Error>, _>`
help: consider using `Result::expect` to unwrap the `std::result::Result<std::result::Result<ConnectionCredentials, std::io::Error>, std::io::Error>` value, panicking if the value is a `Result::Err`
    |
165 |         .await.expect("REASON")
    |               +++++++++++++++++

```

--- a/zbus/src/connection/socket/tcp.rs
+++ b/zbus/src/connection/socket/tcp.rs
@@ -121,7 +121,7 @@ impl ReadHalf for tokio::net::tcp::OwnedReadHalf {
             move || win32_credentials_from_addr(&peer_addr),
             "peer credentials",
         )
-        .await
+        .await?
     }
 
     #[cfg(not(windows))]
@@ -162,7 +162,7 @@ impl WriteHalf for tokio::net::tcp::OwnedWriteHalf {
             move || win32_credentials_from_addr(&peer_addr),
             "peer credentials",
         )
-        .await
+        .await?
     }
 }
 

